test/integration_test/portworx_diag_test.go

```diff
@@ -0,0 +1,910 @@
+		require.NoError(t, ci_utils.DeleteObjects([]runtime.Object{testSTC0, testSTC1}))
+	}
+}
+
+func DiagCollection_SecondDiagShouldWait(tc *types.TestCase) func(*testing.T) {
+	return func(t *testing.T) {
+		testSpec := tc.TestSpec(t)
+		dtc, ok := testSpec.(*diagTestSpec)
+		require.True(t, ok)
+
+		firstDiag := dtc.DiagToCreate.DeepCopy()
+		firstDiag.Name += "-first"
+
+		secondDiag := dtc.DiagToCreate.DeepCopy()
+		secondDiag.Name += "-second"
+
+		// 1. Create the first PortworxDiag object, validate that it eventually goes to In Progress
+		firstDiag, err := operatorops.Instance().CreatePortworxDiag(firstDiag)
+		require.NoError(t, err)
+
+		out, err := task.DoRetryWithTimeout(validatePortworxDiagInPhase(firstDiag, []string{portworxv1.DiagStatusInProgress}), time.Minute*1, time.Second*10)
+		require.NoError(t, err, "Failed to wait for first Portworx Diag to be in progress")
+		firstDiag = out.(*portworxv1.PortworxDiag)
+
+		// 2. Create a second PortworxDiag object, validate that it goes to pending
+		secondDiag, err = operatorops.Instance().CreatePortworxDiag(secondDiag)
+		require.NoError(t, err)
+
+		out, err = task.DoRetryWithTimeout(validatePortworxDiagInPhase(secondDiag, []string{portworxv1.DiagStatusPending}), time.Minute*1, time.Second*10)
+		require.NoError(t, err, "Failed to wait for second Portworx Diag to be pending")
+		secondDiag = out.(*portworxv1.PortworxDiag)
+
+		// 3. Wait for first PortworxDiag to complete
+		out, err = task.DoRetryWithTimeout(validatePortworxDiagInPhase(firstDiag, []string{portworxv1.DiagStatusCompleted}), time.Minute*5, time.Second*15)
+		require.NoError(t, err)
+		firstDiag = out.(*portworxv1.PortworxDiag)
+
+		// 4. Validate cleanup of the first PortworxDiag (but don't yet delete)
+		_, err = task.DoRetryWithTimeout(validateDiagPodsDeleted(), time.Minute, time.Second*15)
+		require.NoError(t, err, "Failed to wait for PortworxDiag pods to be deleted")
+
+		// 5. Validate that the second PortworxDiag is in progress
+		out, err = task.DoRetryWithTimeout(validatePortworxDiagInPhase(secondDiag, []string{portworxv1.DiagStatusInProgress}), time.Minute*1, time.Second*10)
+		require.NoError(t, err, "Failed to wait for second Portworx Diag to be in progress")
+		secondDiag = out.(*portworxv1.PortworxDiag)
+
+		// 6. Clean up (don't bother waiting for the second diag to finish)
+		err = operatorops.Instance().DeletePortworxDiag(secondDiag.Name, secondDiag.Namespace)
+		require.NoError(t, err)
+		err = operatorops.Instance().DeletePortworxDiag(firstDiag.Name, firstDiag.Namespace)
+		require.NoError(t, err)
+	}
+}
+
+func DiagCollection_StopNodes(tc *types.TestCase) func(*testing.T) {
+	return func(t *testing.T) {
+		testSpec := tc.TestSpec(t)
+		dtc, ok := testSpec.(*diagTestSpec)
+		require.True(t, ok)
+
+		// TODO: any other conditions to try?
+		stoppedNode := ci_utils.AddLabelToRandomNode(t, "px/service", "stop")
+		// disabledNode := ci_utils.AddLabelToRandomNode(t, "px/service", "disable")
```

Comment:
Let's remove the commented lines. We don't need to test disable as it would give the same behavior.